src/lib.rs

```diff
@@ -191,51 +200,61 @@ pub struct IdentityIssuanceRequestParameters {
     pub ars_infos: HashMap<u32, AnonymityRevokerInfo>,
     #[serde(rename = "arThreshold")]
     pub ar_threshold: u8,
+    #[derivative(Debug(format_with = "redacted_format"))]
     #[serde(rename = "prfKey")]
     pub prf_key_hex: String,
+    #[derivative(Debug(format_with = "redacted_format"))]
     #[serde(rename = "idCredSec")]
     pub id_cred_sec_hex: String,
+    #[derivative(Debug(format_with = "redacted_format"))]
     #[serde(rename = "blindingRandomness")]
     pub blinding_randomness_hex: String,
 }
 
 /// UniFFI compatible bridge to [`IdentityRecoveryRequestInput`],
 /// providing the implementation of the UDL declaration of the same name.
 /// The translation is performed using Serde.
-#[derive(Clone, Debug, Serialize)]
+#[derive(Clone, Derivative, Serialize)]
+#[derivative(Debug)]
 pub struct IdentityRecoveryRequestParameters {
     #[serde(rename = "ipInfo")]
     pub ip_info: IdentityProviderInfo,
     #[serde(rename = "globalContext")]
     pub global_context: GlobalContext,
     #[serde(rename = "timestamp")]
     pub timestamp: u64,
+    #[derivative(Debug(format_with = "redacted_format"))]
     #[serde(rename = "idCredSec")]
     pub id_cred_sec_hex: String,
 }
 
 /// UniFFI compatible bridge to [`concordium_base::id::types::IpInfo<concordium_base::id::constants::IpPairing>`],
 /// providing the implementation of the UDL declaration of the same name.
 /// The translation is performed using Serde.
-#[derive(Clone, Debug, Serialize)]
+#[derive(Clone, Derivative, Serialize)]
+#[derivative(Debug)]
 pub struct IdentityProviderInfo {
     #[serde(rename = "ipIdentity")]
     pub identity: u32,
     #[serde(rename = "ipDescription")]
     pub description: Description,
+    #[derivative(Debug(format_with = "redacted_format"))]
     #[serde(rename = "ipVerifyKey")]
     pub verify_key_hex: String,
+    #[derivative(Debug(format_with = "redacted_format"))]
     #[serde(rename = "ipCdiVerifyKey")]
     pub cdi_verify_key_hex: String,
 }
 
 /// UniFFI compatible bridge to [`concordium_base::id::types::GlobalContext<concordium_base::id::constants::ArCurve>`],
 /// providing the implementation of the UDL declaration of the same name.
 /// The translation is performed using Serde.
-#[derive(Clone, Debug, Serialize)]
+#[derive(Clone, Derivative, Serialize)]
+#[derivative(Debug)]
 pub struct GlobalContext {
     #[serde(rename = "onChainCommitmentKey")]
     pub on_chain_commitment_key_hex: String,
+    #[derivative(Debug(format_with = "redacted_format"))]
```

Comment:
Are these meant to only mark things that are secret? Or it here because there are a lot of generators?

Reply:
I suppose since it is verbose (you mention secret or verbose in the description).